\documentclass{article}
\usepackage[utf8]{inputenc}
\usepackage{subfiles}
\usepackage[margin=1in]{geometry}
\usepackage{amsmath}
\usepackage{amssymb}
\usepackage{graphicx}
\usepackage{tikz}
\usepackage{subcaption}
\usepackage[font=small, width=\textwidth,margin=3em]{caption}
\usepackage{cite}

\newcommand{\norm}[1]{\left\lVert#1\right\rVert}


%\title{SCarborSNV: Efficient phylogeny-aware single nucleotide variant detection for single cells}
%\author{Christopher Oldham}
%\date{January 2019}

\begin{document}

%\maketitle
\subfile{titlepage}
\subfile{sections/abstract}

\section{Introduction}
\subfile{sections/introduction}

\section{Methods}
\subfile{sections/methods}

\section{Results}
Include complexity analysis
\section{Discussion}

Two methods for leaves affected: power law and uniform branches. Uniform branches (sciphi) used for sampled vs power law for considering whole tumour. Cannot use power law for sample? Why? Why not?\\
Overall question: is $O(nm^3)$ good enough at all stages? (No!) asymptotically only beats sci$\phi$ by $\log(m)$... Ofc NJ is $O(m^3)$ but not $nm^3$ since you build a single tree from info on all sites! So my alg could be bounded by $O(nm^2)$?? Computing distances in $O(nm^2)$... Could have an $O(nm^2)$ algorithm! 

%TODO near eqns 13, 14 instead of reffering to phasing so heavily, instead beter to show probability of an unphased genotype vector given \sigma, refer to Li, statistical framework for SNP calling...



\newpage
\bibliography{SNVite}{}
\bibliographystyle{unsrt}

\newpage
\appendix
\subfile{sections/appendix}


\end{document}

%After done with algorithm: simulator! Real data results, efficiency comparison, introduction and discussion... Poster? Style, change from we to factual voice? dbSNP and missing data?? Account for different read depths? likelihood for single read much greater than that for multi: so maybe normalize each?
