\documentclass[../../main.tex]{subfiles}

\begin{document}
%TODO XXX priors before this not finally checked for writing

Having found the prior probability distribution for the alternate allele count across all cells at a locus, we now turn to the likelihoods of individual cells at a locus.
The (unphased) genotype $g\in\{0,1,2\}$ for cell $j$ at locus $i$ is equivalent to the specific alternate allele count for that cell.
At this stage of the calculation we consider all sites and cells independently.
\subsubsection*{Homozygous genotypes}
For the cell likelihood $\Lambda(g) = P(d_{ij}\,\mid\,g)$ we assume the error on each read to be independent of the other reads.
Hence if cell $j$ has $l$ reads covering locus $i$ we have:
\begin{equation}
    P(d_{ij}\mid g) = \prod_{k=1}^l P(d_{ijk} \mid g)
\end{equation}
Note $d_{ij}=(\vec{r},\vec{e})$, where $\vec{r}$ are the $l$ reads for this cell at this locus.
$\vec{e}$ are the associated probabilities of read error, which are directly computed from the the phred quality scores given by the sequencing machine.

Marginalizing on sequencing error:
\begin{equation*}
    P(d_k \mid g) =  P(r_k,\,se \mid g) + P(r_k,\,\neg se \mid g)
\end{equation*}
\begin{itemize}
    \item Since errors can occur during amplification or sequencing, we model an "intermediate allele", denoted $\beta$ that is amplified from the original nucleotide with some probability of error~\cite{monovar}. Trivially:
    \begin{equation*}
        P(r_k,\,\neg se \mid g) = P(r_k\mid \neg se ,\, g)(1-e_k)=P(\beta_k=r_k\mid g)(1-e_k) 
    \end{equation*}
    We similarly see:
    \begin{equation*}
        P(r_k,\, se \mid g) = P(r_k\mid se,\, g)e_k
    \end{equation*}
    Furthermore:
    \begin{equation*}
        P(r_k\mid se,\, g) = P(r_k\mid \beta_k\neq r_k,\,se,\,g) P(\beta_k\neq r_k \mid se,\,g)
    \end{equation*}
     Assuming (?) an error in sequencing the intermediate allele could produce any of the other three alleles with equal probability we find $P(r_k\mid \beta_k \neq r_k,\,se,g)=1/3$. Since the amplification of $\beta$ is unaffected by sequencing $P(\beta_k\neq r_k \mid se,\,g)=P(\beta_k\neq r_k\mid g)=1-P(\beta_k=r_k\mid g)$. We therefore have:
     \begin{equation*}
         P(r_k, se \mid g) = e_k \frac{1}{3}\left[1-P(\beta_k=r_k\mid g)\right]
     \end{equation*}
     \item Finally the likelihood $P(D_{ij}\mid g)$ for cell at a locus for a homozygous genotype is:
     \begin{equation}
          P(D_{ij}\mid g) = \prod_{k=1}^l \left[ (1-e_k)P(\beta_k=r_k\mid g) + e_k \frac{1}{3} (1-P(\beta_k=r_k\mid g)) \right]
     \end{equation}
\end{itemize}
\subsubsection*{Heterozygous genotypes and allelic dropout}
\begin{itemize}
     \item For the heterozygous case, we must account for the possibility of allelic dropout (ADO)~\cite{monovar,sciphi}. Therefore:
     \begin{equation*}
         P(D_{ij}\mid g=1) = P(D_{ij}, \text{ADO} \mid g=1)+ P(D_{ij}, \neg \text{ADO}\mid g=1)
     \end{equation*}
     Letting $P_{ADO}$ be the probability of an a dropout event, this expands to:
     \begin{equation*}
         P(D_{ij}\mid g=1) = P_{ADO}P(D_{ij}\mid \text{ADO},\, g=1) + (1-P_{ADO})P(D_{ij} \mid \neg \text{ADO},\, g=1)
     \end{equation*}
     In the result of an allelic dropout from a heterozygous locus, only one allele will remain after the amplification process and hence the likelihood $P(D_{ij}\mid \text{ADO},\, g=1)$ will resemble the homozygous case. We assume allelic dropout can affect either  allele with equal probability and hence:
     \begin{equation*}
         P(D_{ij}\mid \text{ADO},\, g=1) =\frac{1}{2}P(D_{ij}\mid g=0) + \frac{1}{2}P(D_{ij}\mid g=2)
     \end{equation*}
     For the case without allelic dropout, the form of the likelihood is identical to the homozygous case:
     \begin{equation*}
           P(D_{ij}\mid \neg\text{ADO},\,g=1) = \prod_{k=1}^l \left[ (1-e_k)P(\beta_k=r_k\mid g=1) + e_k \frac{1}{3} (1-P(\beta_k=r_k\mid g=1)) \right]
     \end{equation*}
\end{itemize}
%NOTE: for thesis good to include all calculation but for paper should probably just cite monovar

\end{document}
