\documentclass[../../main.tex]{subfiles}

\begin{document}
The most accurate phylogenetic structure of the sampled tumour cells could be found by searching through the entire tree space and finding a tree that maximizes likelihood or posterior probability. If $s$ mutant sites are called in the previous step there are $(2m-3)!!(2m-1)^s$ trees in the search space making this approach infeasible, leading a previous phylogeny aware approach to adopt a more efficient Markov chain Monte Carlo (MCMC) algorithm. This more efficient approach results in an overall asymptotic complexity of $O(nm^3\log(m))$~\cite{sciphi}. The Monovar algorithm has an overall asymptotic complexity of $O(nm^3)$.

%For now, build tree based just on cell likelihoods or basic posteriors. Later update this to posteriors based on site posteriors. The ``proper" way can be found in monovar under ``Genotyping of sigle cells". Another way could be to sum over sigma with some simple formula like binomial (implies certain non existant independence). Is there a way to use the DP values? frequency based?\\
%TODO implement real posteriors?
We use a simple neighbour-joining algorithm to infer a cell phylogeny based on the sites called as variant candidates. While this approach still has an asymptotic complexity of $O(nm^3)$ we expect it will yeild results faster even than Monovar on real and simulated data.
%TODO remove the above if making the switch to monovar style posteriors.
While Monovar determines cell genotype posteriors by simultaneously considering the cell in question and the probability of all other cells having allele count $\sigma-g$, for efficiency we simply use a binomial model:
\begin{equation*}
P(g_{ij}\mid D_i) = \sum_{\sigma=1}^{2m}\binom{2}{g}\left(\frac{\sigma}{2m}\right)^g\left(1-\frac{\sigma}{2m}\right)^{2-g}P(\sigma\mid D_i)
\end{equation*}
where $P(\sigma\mid D_i)$ can be determined using Bayes' formula using the memoized values computed for Equation~\ref{eq:sitebayes}. Next we define a pairwise value $\overline{p}$, the expected frequency with which nucleotides differ between two cells $a$ and $b$:
\begin{equation*}
\overline{p} = \frac{1}{2n}\sum_{i=1}^n\left[ \sum_{|g_{ia}-g_{ib}|=2} 2P(g_{ia})P(g_{ib}) + \sum_{|g_{ia}-g_{ib}|=1} P(g_{ia})P(g_{ib})\right]
\end{equation*}
%TODO nb could do monovar style trick here. instead of independent posteriors do pairwise posteriors... Suuuuper complex tho? m^4? m(m-1) pairs, O(m^2) to genotype each...
Note we assume that if two cells have the same alternate allele count at a locus they have the same phased genotype at that locus. We then compute a distance inspired by the Jukes-Cantor distance:
\begin{equation}
d = \log \left(1-\frac{4}{3}\,\overline{p}\right)
\end{equation}
Included with the biological cells is a false cell with homozygous reference genotype which will be used to root the tree. After computing $d$ for all pairs of cells, we implement a simple neighbor-joining algorithm based thereon~\cite{BSA}.



\end{document}
