\documentclass[../main.tex]{subfiles}

\begin{document}

Inferring the various genetic mutations present in a sample of phylogenetically related single cells is a complicated task, especially in light of the low quality data generated by SCS methods.
As such, previous successful methods have pooled information from across all cells in a sample to more accurately distinguish between bona fide mutations and noise.
The commonly used Monovar algorithm pools the information from across cells by marginalizing on the variable total alternate allele count across all cells at each locus, using statistical methods derived from population genetics models~\cite{monovar}.
A more recent approach, Sci$\Phi$, has found further success by leveraging the fact that the sampled cells are phylogenetically related.
By first gaining some understanding of this ancestral relationship, Sci$\Phi$ has been able to improve calling accuracy with similar algorithmic complexity to Monovar~\cite{sciphi}.

The SCarborSNV algorithm presented here intends to follow on in the direction pioneered by Sci$\Phi$, using phylogenetic inference to distinguish between real mutations and SCS noise.
SCarborSNV shows similar calling accuracy to state-of-the-art tools on simulated datasets, with asymptotically improved time complexity in the number of cells sampled, from $O(m^3)$ to $O(m^2)$.

\subsection{Cells, Mutation and Intra-tumour Heterogeneity}
\subfile{sections/subintro/biological}

\subsection{High Throughput Sequencing and Bulk Approaches}
\subfile{sections/subintro/HTSbulk}

\subsection{Single Cell Sequencing}
\subfile{sections/subintro/SCS}

\subsection{Simplifications and Problem Scope}
\subfile{sections/subintro/simplifications}

\subsection{Phylogenetic Inference}
\subfile{sections/subintro/inference}
 
\end{document}
