\documentclass[../main.tex]{subfiles}

\begin{document}

\subfile{sections/submethods/method_introduction}

%\subsection{Preprocessing and parallelism}
%\subfile{sections/submethods/preprocess}
%TODO write an appendix for preprocessing pipeline

\subsection{Mutated site priors}
\subfile{sections/submethods/mutated_priors}

\subsection{Welltype site priors}
\subfile{sections/submethods/well_priors}

\subsection{Cell genotype likelihoods}
\subfile{sections/submethods/cell_likelihoods}

\subsection{Alternate allele count likelihoods}
\subfile{sections/submethods/lambda_likelihoods}

\subsection{Pairwise distances and candidate loci}
\subfile{sections/submethods/candidates_pairwise}

\subsection{Building a cell phylogeny}
\subfile{sections/submethods/phylo1}

%TODO does binomial model account for large number of potential homozygotes?

\subsection{Genotyping single cells}
\subfile{sections/submethods/inference}

\subsection{Additional computational methods}
Some likelihoods used in SCarborSNV may be the product of many small probabilities, especially if read depth is high.
Therefore, despite the use of extended precision \texttt{long double} types, there is a possibility of floating point underflow occurring when manipulating and storing these values.
Therefore all of SCarborSNV's calculations are done in log-space.

Sums of probabilities or likelihoods would require converting back from log-space and potential underflow.
Therefore SCarborSNV has implemented LogSumExp functions to minimize potential underflow in these situations.


\end{document}
